src/Project/Compress.py

```diff
@@ -91,10 +91,11 @@ def move_extracted_song_to_temp(self):
                 )
 
                 audio_type = "Scratch" if mp3.parent.name == "Media" else "#MIXDOWN#"
-                Slack.send_link(
-                    link_name = f'{audio_type} for {self.entry.name}, "{mp3.name}"',
-                    ID        = mp3_id
-                )
+
+                link_title = f'{audio_type} for {self.entry.name}, "{mp3.name}"'
+                url        = Drive.file_url(mp3_id)
+
+                Discord().post_link(link_title, url)
```

Comment:
This one I do have strong feelings about. The thing I really want to avoid is having implicit behavior in the constructor. In my opinion, the constructor shouldn't really _do_ anything other than return an instance of the class. My understanding of good object-orientated design is that an object's behavior should be defined in its methods. In my experience, this tends to work out well over the long-term, and the benefits outweigh the drawbacks (namely, more verbose and ugly code). I can definitely elaborate on some of the specific benefits if you're not fully sold on this.

All of that said, I think we can still improve this. Take a look at [this commit](https://github.com/vindennl48/LOFSongManager/pull/10/commits/cd8798947b47eb06f52d2d5ce1fbae08b2c6270a). Instead of having to instantiate a new object with `Discord()` and then chaining a `post_*` method every time, I thought it would be cleaner to have a single class method that would accept a dict of options and data. The result is that notifications are created by calling `Discord.notify({ ... })`.

It's still not perfect, but I think it makes for a better interface, and it feels more natural than how it was before. What do you think?

Reply:
I think we are discussing two different topics.  My comment here was about these two options:
`
link_title = f'{audio_type} for {self.entry.name}, "{mp3.name}"'
url           = Drive.file_url(mp3_id)

Discord().post_link(link_title, url)
`

as opposed to:

`
Discord().post_link(
    f'{audio_type} for {self.entry.name}, "{mp3.name}"',
    Drive.file_url(mp3_id)
)
`

The topic you are referring to I mentioned in another comment.  I will address it there.